core/src/main/java/com/graphhopper/routing/profiles/MappedDecimalEncodedValue.java

```diff
@@ -63,18 +65,53 @@ private int toInt(double val) {
         return (int) Math.round(val / precision);
     }
 
+    int findClosestIndex(int val) {
+        int res = Arrays.binarySearch(toValueMapArray, val);
+        if (res >= 0)
+            throw new IllegalStateException("Cannot happen as toStorageMap.getOrDefault should have caught this case");
+
+        // TODO NOW rounding up from e.g. max_height=10 to 11 might be ugly. Not sure how or what to do about it or if we should make it even configurable.
+        res = -res;
+        double delta1 = res < 2 ? Double.POSITIVE_INFINITY : val - toValueMapArray[res - 2],
+                delta2 = res >= toValueMapArray.length ? Double.POSITIVE_INFINITY : toValueMapArray[res - 1] - val;
```

Comment:
It might be worth to add a comment here explaining what is done. The two inline ifs are a bit hard to read :).